cli/medperf/entities/cube.py

```diff
@@ -122,81 +121,77 @@ def get(cls, cube_uid: str) -> "Cube":
             return local_cube[0]
 
         meta = comms.get_cube_metadata(cube_uid)
-        cube = cls(cube_uid, meta, "")
+        cube = cls(meta)
         cube.download()
-        local_hashes = {
-            "additional_files_tarball_hash": cube.additional_hash if cube.additional_hash else "",
-            "image_tarball_hash": cube.image_tarball_hash if cube.image_tarball_hash else "",
-        }
-        save_cube_metadata(meta, local_hashes)
+        cube.write()
         return cube
 
     def download(self):
         """Downloads the required elements for an mlcube to run locally.
         """
         comms = config.comms
-        # Backwards compatibility for cubes with
-        # tarball_url instead of additional_files_tarball_url
-        old_files = "tarball_url"
-        old_hash = "tarball_hash"
-        add_files = "additional_files_tarball_url"
-        add_hash = "additional_files_tarball_hash"
-        meta = self.meta
+        ui = config.ui
         cube_uid = self.uid
-        if old_files in meta:
-            meta[add_files] = meta[old_files]
-            meta[add_hash] = meta[old_hash]
-        cube_path = comms.get_cube(meta["git_mlcube_url"], cube_uid)
-        params_path = None
-        additional_path = None
-        additional_hash = None
-        image_path = None
-        image_tarball_hash = None
-        if "git_parameters_url" in meta and meta["git_parameters_url"]:
-            url = meta["git_parameters_url"]
-            params_path = comms.get_cube_params(url, cube_uid)
-        if add_files in meta and meta[add_files]:
-            url = meta[add_files]
+        self.cube_path = comms.get_cube(self.git_mlcube_url, cube_uid)
+        local_additional_hash = ""
+        local_image_hash = ""
+        if self.git_parameters_url:
+            url = self.git_parameters_url
+            self.params_path = comms.get_cube_params(url, cube_uid)
+        if self.additional_files_tarball_url:
+            url = self.additional_files_tarball_url
             additional_path = comms.get_cube_additional(url, cube_uid)
-            additional_hash = get_file_sha1(additional_path)
+            if not self.additional_hash:
+                # log interactive ui only during submission
+                ui.text = "Generating additional file hash"
```

Comment:
I'm not sure how I feel about an entity directly interacting with the UI. If someone in the future wants to use medperf entities as an API, they'll encounter ui prints that could disrupt their workflows. I just realized this, and I'm aware we've been probably doing the same for a while with `pretty_error`s called by entities. 

It would be great to look into separating concerns here in the future.
- Entities are in charge of doing the hard work, and are unaware of an user interacting with them
- Commands are in charge of giving the user ways of interacting with entities, and therefore should provide user feedback.

Reply:
yeah.. maybe splitting up `cube.download` in the future would be a solution, given that also it is going to be extended for `mlcube.yaml` and `parameters.yaml` hash calculations as well.